### PR TITLE
Complete sudoers pw

### PR DESCRIPTION
**Pull request by ttheune**
Tested with updated testing tools.

### CODE REVIEW

File: linux/amzn/2014.09/foxpass_setup.py [hunk 1]
```diff
@@ -139,38 +146,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
...
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
+    if not require_sudoers_pw:
+        os.system("sed -i 's/^%wheel\tALL=(ALL:ALL) ALL/%wheel ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers")
```
Comment (aren) [inline linux/amzn/2014.09/foxpass_setup.py]: this is different than the previous `sed` line. on purpose?

File: linux/amzn/2016.03/foxpass_setup.py [hunk 2]
```diff
@@ -138,40 +145,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
...
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
-
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
+    if not require_sudoers_pw:
+        os.system("sed -i 's/^%wheel\tALL=(ALL:ALL) ALL/%wheel ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers")
```
Comment (aren) [inline linux/amzn/2016.03/foxpass_setup.py]: this is different than the previous `sed` line. on purpose?

File: linux/centos/7/foxpass_setup.py [hunk 3]
```diff
@@ -138,38 +145,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
...
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
+    if not require_sudoers_pw:
+        os.system("sed -i 's/^%wheel\tALL=(ALL:ALL) ALL/%wheel ALL=(ALL:ALL) NOPASSWD:ALL/' /etc/sudoers")
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: this is different than the previous `sed` line. on purpose?

File: linux/amzn/2016.03/foxpass_setup.py (hunk 2 above)
Comment (aren) [inline linux/amzn/2016.03/foxpass_setup.py]: this is different than the previous `sed` line. on purpose?

File: linux/amzn/2014.09/foxpass_setup.py [hunk 4]
```diff
@@ -139,38 +146,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
...
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/amzn/2014.09/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/amzn/2016.03/foxpass_setup.py [hunk 5]
```diff
@@ -138,40 +145,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
...
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
-
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/amzn/2016.03/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/centos/7/foxpass_setup.py [hunk 6]
```diff
@@ -138,38 +145,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
...
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/centos/7/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/debian/8/foxpass_setup.py [hunk 7]
```diff
@@ -163,68 +169,76 @@ def write_nslcd_conf(uris, basedn, binddn, bindpw, threads, idle_timelimit):
...
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/debian/8/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/ubuntu/12.04/foxpass_setup.py [hunk 8]
```diff
@@ -164,68 +172,76 @@ def write_nslcd_conf(uris, basedn, binddn, bindpw, threads, idle_timelimit):
...
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/ubuntu/12.04/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/ubuntu/14.04/foxpass_setup.py [hunk 9]
```diff
@@ -175,68 +180,76 @@ def write_nslcd_conf(uris, basedn, binddn, bindpw, threads, idle_timelimit):
...
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/ubuntu/14.04/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/ubuntu/16.04/foxpass_setup.py [hunk 10]
```diff
@@ -175,68 +181,76 @@ def write_nslcd_conf(uris, basedn, binddn, bindpw, threads, idle_timelimit):
...
         with open('/etc/sudoers', 'a') as w:
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL) NOPASSWD:ALL'.format(sudo=sudoers))
-        os.system('chmod 440 /etc/sudoers.d/95-foxpass-sudo')
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/ubuntu/16.04/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/ubuntu/17.04/foxpass_setup.py [hunk 11]
```diff
@@ -213,16 +213,15 @@ def fix_nsswitch():
...
             w.write('\n#includedir /etc/sudoers.d\n')
     if not os.path.exists('/etc/sudoers.d'):
         os.system('mkdir /etc/sudoers.d && chmod 750 /etc/sudoers.d')
     if not os.path.exists('/etc/sudoers.d/95-foxpass-sudo'):
         with open('/etc/sudoers.d/95-foxpass-sudo', 'w') as w:
-            w.write('# Adding Foxpass group to sudoers\n\
-            %{sudo} ALL=(ALL:ALL){command}'.format(sudo=sudoers,
-                                                   command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
+            w.write('# Adding Foxpass group to sudoers\n%{sudo} ALL=(ALL:ALL){command}'.
+                    format(sudo=sudoers, command='NOPASSWD:ALL' if not require_sudoers_pw else 'ALL'))
```
Comment (aren) [inline linux/ubuntu/17.04/foxpass_setup.py]: should be flipped:

`command='ALL' if require_sudoers_pw else 'NOPASSWD:ALL'`

File: linux/amzn/2014.09/foxpass_setup.py [hunk 12]
```diff
@@ -139,38 +146,38 @@ def configure_sssd(bind_dn, bind_pw, backup_ldaps):
 
 
 def augment_sshd_config():
-    if not file_contains('/etc/ssh/sshd_config', '^AuthorizedKeysCommand'):
+    if not file_contains('/etc/ssh/sshd_config', r'^AuthorizedKeysCommand '):
```
Comment (aren) [inline linux/amzn/2014.09/foxpass_setup.py]: trailing space on purpose?

File: linux/centos/7/foxpass_setup.py (hunk 3 above)
Comment (ttheune) [reply linux/centos/7/foxpass_setup.py]: Copy Pasta error
Got through testing because of the foxpass_sudoers group (need to update that)

**Pull request merged**